File: frontend/lib/models/card_account.dart
class CardAccount {
  final int id;
  final int showroomId;
  final String? showroomName;
  final String bankName;
  final String lastFour;
  final double currentBalance;
  final bool isActive;
  final String? createdAt;

  CardAccount({
    required this.id,
    required this.showroomId,
    this.showroomName,
    required this.bankName,
    required this.lastFour,
    required this.currentBalance,
    required this.isActive,
    this.createdAt,
  });

  factory CardAccount.fromJson(Map<String, dynamic> json) {
    return CardAccount(
      id: json['id'] as int,
      showroomId: json['showroom_id'] as int,
      showroomName: json['showroom']?['name'] as String?,
      bankName: json['bank_name'] as String,
      lastFour: json['last_four'] as String,
      currentBalance:
          double.tryParse(json['current_balance'].toString()) ?? 0.0,
      isActive: json['is_active'] == true || json['is_active'] == 1,
      createdAt: json['created_at'] as String?,
    );
  }

  Map<String, dynamic> toJson() => {
        'id': id,
        'showroom_id': showroomId,
        'bank_name': bankName,
        'last_four': lastFour,
        'current_balance': currentBalance,
        'is_active': isActive,
      };

  String get maskedNumber => '•••• $lastFour';

  String get displayLabel => '$bankName •••• $lastFour';

  String dropdownLabel(String? sName) =>
      '${sName ?? showroomName ?? ''} — $bankName •••• $lastFour';
}

File: ios/Runner/Models/CardAccount.swift
import Foundation

struct CardAccount: Decodable, Identifiable {
    let id: Int
    let showroomId: Int
    let showroomName: String?
    let bankName: String
    let lastFour: String
    let currentBalance: Double
    let isActive: Bool
    let createdAt: String?

    var maskedNumber: String { "•••• \(lastFour)" }
    var displayLabel: String { "\(bankName) •••• \(lastFour)" }

    func dropdownLabel(_ sName: String?) -> String {
        "\(sName ?? showroomName ?? "") — \(bankName) •••• \(lastFour)"
    }

    enum CodingKeys: String, CodingKey {
        case id
        case showroomId      = "showroom_id"
        case showroom
        case bankName        = "bank_name"
        case lastFour        = "last_four"
        case currentBalance  = "current_balance"
        case isActive        = "is_active"
        case createdAt       = "created_at"
    }

    // Nested showroom object for showroomName
    private struct ShowroomNested: Decodable { let name: String }

    init(from decoder: Decoder) throws {
        let c = try decoder.container(keyedBy: CodingKeys.self)
        id            = try c.decode(Int.self, forKey: .id)
        showroomId    = try c.decode(Int.self, forKey: .showroomId)
        bankName      = try c.decode(String.self, forKey: .bankName)
        lastFour      = try c.decode(String.self, forKey: .lastFour)
        createdAt     = try? c.decode(String.self, forKey: .createdAt)

        let balStr    = try c.decode(AnyCodable.self, forKey: .currentBalance)
        currentBalance = Double("\(balStr.value)") ?? 0

        if let b = try? c.decode(Bool.self, forKey: .isActive) {
            isActive = b
        } else {
            isActive = (try? c.decode(Int.self, forKey: .isActive)) == 1
        }
        showroomName = (try? c.decode(ShowroomNested.self, forKey: .showroom))?.name
    }
}
